addons/im_livechat/models/res_partner.py

```diff
@@ -4,14 +4,29 @@
 
 from odoo import api, models, fields, _
 from odoo.addons.mail.tools.discuss import Store
-
+from odoo.tools.misc import OrderedSet
 
 class ResPartner(models.Model):
     """Update of res.partner class to take into account the livechat username."""
     _inherit = 'res.partner'
 
     user_livechat_username = fields.Char(compute='_compute_user_livechat_username')
 
+    def _get_search_for_channel_invite_term_domain(self, search_term):
+        return super()._get_search_for_channel_invite_term_domain(search_term) + [
+            [("user_ids.livechat_lang_ids.name", "ilike", search_term)],
+            [("user_ids.res_users_settings_ids.livechat_expertise_ids.name", "ilike", search_term)]
+        ]
```

Comment:
Should use new `Domain` class especially when combining them like that across methods.